packages/core/src/types.ts

```diff
@@ -561,8 +561,15 @@ type OUTGOING_EVENT_PAYLOADS = {
   outsideCanvasClick: OutsideCanvasClickPayload;
   sdkFeatures: SDKFeaturesPayload;
   REQUEST_ENTITIES: RequestEntitiesPayload;
+  canvasGeometryUpdated: {
+    size: { width: number; height: number };
+    nodes: Record<string, { coordinates: Pick<DOMRect, 'x' | 'y' | 'width' | 'height'> }>;
+    sourceEvent: CanvasGeometryUpdateSourceEvent;
+  };
 };
 
+export type CanvasGeometryUpdateSourceEvent = 'resize' | 'mutation' | 'manual';
```

Comment:
tbh `manual` sounds a bit misleading, since we don't actually trigger it by hand. maybe something like `treeUpdate`, but i'm not even sure if we need to trigger it there, since it will still fire on dom changes

Reply:
You're right, the name is misleading. As this is more like a general "others" scenario, I'm using instead the `undefined` value to not give it an explicit label.

Reply:
We don't need it necessarily. I thought it might be helpful during debugging to surface the reason for SDK messages being fired. Would you prefer leaving it out?